### PR TITLE
Refactoring `ConvModule` by removing `norm_cfg`

### PR DESCRIPTION
**Pull request by sungchul1**
### Summary

This PR includes:
- [x] Update `build_norm_layer`
    - This function can handle various format of `normalization`:

        | given format |  | returned format |  |
        | -- | -- | -- | -- |
        | `tuple[norm_name, norm_layer]` | generated by `build_norm_layer` before | `tuple[norm_name, nn.Module]`  | returned as is |
        | `nn.Module` | predefined module | `("", nn.Module)` | returned with empty name string |
        | `nn.BatchNorm2d` | `type` is only given | `(norm_name, nn.Module)` | built and returned with default settings and given `num_features` |
        | `partial(nn.BatchNorm2d, eps=1)` | `partial` for norm layer is given | `(norm_name, nn.Module)` | built and returned with custom settings and given `num_features` |
        | `partial(build_norm_layer(nn.BatchNorm2d, requires_grad=True))` | `partial` of `build_norm_layer` for arguments of `build_norm_layer` | `(norm_name, nn.Module)` | built and returned with custom settings and given `num_features` |

- [x] Replace `norm_cfg` with `normalization`
- [x] Restore `build_activation_layer`
- [x] Update `ConvModule` to use pre-instantiated normalization module and activation module
- [x] Update `activation` docstring

### Checklist

- [ ] I have added unit tests to cover my changes.​
- [ ] I have added integration tests to cover my changes.​
- [ ] I have ran e2e tests and there is no issues.
- [x] I have added the description of my changes into CHANGELOG in my target branch (e.g., [CHANGELOG](https://github.com/openvinotoolkit/training_extensions/blob/develop/CHANGELOG.md) in develop).​
- [ ] I have updated the documentation in my target branch accordingly (e.g., [documentation](https://github.com/openvinotoolkit/training_extensions/tree/develop/docs) in develop).
- [ ] I have [linked related issues](https://help.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword).

### License

- [ ] I submit _my code changes_ under the same [Apache License](https://github.com/openvinotoolkit/training_extensions/blob/develop/LICENSE) that covers the project.
      Feel free to contact the maintainers if that's a concern.
- [ ] I have updated the license header for each file (see an example below).

```python
# Copyright (C) 2024 Intel Corporation
# SPDX-License-Identifier: Apache-2.0
```

### CODE REVIEW

**Review by harimkang — DISMISSED** [review]

**Comment by sungchul1** [comment]
@sooahleex @harimkang I saw the performance drop for deit_tiny and efficientnet_v2 on hlabel classification as below.
I used only medium set (hlabel_CUB_medium), but classification has been using deterministic seed and its performance was maintained.

| commit | deit_tiny | efficientnet_v2 |
| -- | -- | -- |
| 1a8e10e | 0.93 | 0.97 |
| 2ecaac149 | 0.87 | 0.39 |

If you already know this, feel free to ignore it :)

**Review by harimkang — DISMISSED** [review]

**Review by harimkang — DISMISSED** [review]

**Review by harimkang — APPROVED** [review]

File: src/otx/algo/modules/activation.py [hunk 1]
```diff
@@ -32,3 +35,72 @@ def forward(self, x: Tensor) -> Tensor:
...
+
+
+def build_activation_layer(
+    activation: Callable[..., nn.Module] | nn.Module | None,
+    inplace: bool = True,
+) -> nn.Module | None:
+    """Build activation layer.
+
+    Args:
+        activation (Callable[..., nn.Module]): Activation layer module.
```
Comment (eunwoosh) [inline src/otx/algo/modules/activation.py:84]:
```suggestion
        activation (Callable[..., nn.Module] | nn.Module | None): Activation layer module.
```

File: src/otx/algo/modules/conv_module.py [hunk 2]
```diff
@@ -78,14 +79,15 @@ class ConvModule(nn.Module):
...
-        inplace (bool): Whether to use inplace mode for activation.
-            Default: True.
+        bias (bool | str): If specified as `auto`, it will be decided by the normalization.
+            Bias will be set as True if `normalization` is None, otherwise False.
+            Defaults to "auto".
+        normalization (tuple[str, nn.Module] | None): Tuple of normalization layer name and module
+            generated by ``build_norm_layer``.
+            Defaults to None.
+            TODO (sungchul): enable nn.Module standalone and return it with appropriate name
+        activation (Callable[..., nn.Module] | None): Activation layer module.
```
Comment (eunwoosh) [inline src/otx/algo/modules/conv_module.py:89]:
```suggestion
        activation (Callable[..., nn.Module] | nn.Module | None): Activation layer module.
```

File: src/otx/algo/modules/conv_module.py
Comment (eunwoosh) [inline src/otx/algo/modules/conv_module.py:1]: It seems that `AVAILABLE_ACTIVATION_LIST` and `ACTIVATION_LIST_NOT_SUPPORTING_INPLACE` aren't used anywhere. If then, it's better to remove them.

File: src/otx/algo/modules/conv_module.py [hunk 3]
```diff
@@ -319,25 +302,45 @@ def __init__(
...
+            return build_norm_layer(normalization, num_features=in_channels)
+
         # if norm/activation config of depthwise/pointwise Conv2dModule is not
         # specified, use default config.
-        dw_norm_cfg = dw_norm_cfg or norm_cfg
-        dw_activation_callable = dw_activation_callable or activation_callable
-        pw_norm_cfg = pw_norm_cfg or norm_cfg
-        pw_activation_callable = pw_activation_callable or activation_callable
+        dw_normalization = dw_normalization or deepcopy(normalization)
+        dw_normalization = update_num_features(dw_normalization, in_channels)
```
Comment (eunwoosh) [inline src/otx/algo/modules/conv_module.py:338]: It's just a question. I think it's better to remove `normalization` from arguments and require user to give both `dw_normalization` and `pw_normalization`. How do you think about it?

**Review by eunwoosh — APPROVED** [review]
Thanks for your lots of work :) LGTM. I left some minor comments. Please check it's worth to apply later.

File: src/otx/algo/modules/conv_module.py (hunk 3 above)
Comment (sungchul1) [reply src/otx/algo/modules/conv_module.py:338]: It is necessary when it is unclear whether to use `DepthwiseSeparableConvModule` or `Conv2dModule` like below
https://github.com/openvinotoolkit/training_extensions/blob/a5470fc3665228be8c3fde0bdc0728b11b26c057/src/otx/algo/common/backbones/cspnext.py#L166-L173

File: src/otx/algo/modules/activation.py (hunk 1 above)
Comment (sungchul1) [reply src/otx/algo/modules/activation.py:84]: https://github.com/openvinotoolkit/training_extensions/pull/3868

File: src/otx/algo/modules/conv_module.py (hunk 2 above)
Comment (sungchul1) [reply src/otx/algo/modules/conv_module.py:89]: https://github.com/openvinotoolkit/training_extensions/pull/3868

File: src/otx/algo/modules/conv_module.py
Comment (sungchul1) [reply src/otx/algo/modules/conv_module.py:1]: https://github.com/openvinotoolkit/training_extensions/pull/3868
